remotes/docker/pusher.go

```diff
@@ -132,6 +145,48 @@ func (p dockerPusher) Push(ctx context.Context, desc ocispec.Descriptor) (conten
 		}
 		req.Header.Add("Content-Type", desc.MediaType)
 	} else {
+		mountCandidates := findMatchingOrigins(p.refspec, p.originProvider(desc))
+		for _, mountCandidate := range mountCandidates {
+			ctx, err := contextWithRepositoryScope(ctx, mountCandidate, false, true)
+			if err != nil {
+				// try next candidate
+				continue
+			}
+			i := strings.Index(mountCandidate.Locator, "/")
+			originRepoPath := mountCandidate.Locator[i+1:]
+			req, err = http.NewRequest(http.MethodPost, fmt.Sprintf("%s/?mount=%s&from=%s", p.url("blobs", "uploads"), desc.Digest, originRepoPath), nil)
```

Comment:
Should `mount` and `from` not be added with somethign like:
```go
                q := req.URL.Query()
                q.Add("digest", desc.Digest.String())
                req.URL.RawQuery = q.Encode()
```
(which I stole from further down in this function). It's still more roundabout than I would like but is better than opencoding the query construction I think.